packages/db/prisma/utils/dates.ts

```diff
@@ -8,25 +8,39 @@ export type RawDatePeriod = {
   from?: RawDate;
   to?: RawDate;
 };
+
 export type DatePeriod = {
   from?: string;
   to?: string;
 };
 
-export function normalizeDate(
-  date: dayjs.Dayjs | string | Date,
+export function normalizeDailyDate(
+  date: string | Date | dayjs.Dayjs,
   startOfOrEndOfDay: "startOf" | "endOf" = "endOf"
 ) {
   const date_ = dayjs.isDayjs(date) ? date : dayjs(new Date(date)).utc();
 
   return date_[startOfOrEndOfDay]("day").toISOString();
 }
 
-export function normalizeDatePeriod(datePeriod: RawDatePeriod): DatePeriod {
-  return {
-    from: datePeriod.from && normalizeDate(datePeriod.from, "startOf"),
-    to: datePeriod.to && normalizeDate(datePeriod.to, "endOf"),
+export function normalizeDailyDatePeriod(
+  datePeriod?: RawDatePeriod
+): DatePeriod {
+  if (!datePeriod || (!datePeriod.from && !datePeriod.to))
+    return { to: normalizeDailyDate(dayjs()) };
+
+  const normalizedDatePeriod = {
+    from: datePeriod.from && normalizeDailyDate(datePeriod.from, "startOf"),
+    to: datePeriod.to && normalizeDailyDate(datePeriod.to, "endOf"),
   };
+
+  const { from, to } = normalizedDatePeriod;
+
+  if (from && to && new Date(from) > new Date(to)) {
```

Comment:
At this point, we are sure that both `from` and `to` are defined. Aren't we?

Reply:
No. You can specify a date period without a starting or ending date so you need to check if they're defined